docs/set.rst

```diff
@@ -0,0 +1,487 @@
+
+::
+
+    Set.toDescList :: Set a -> [a]
+    Set.toDescList s = ...
+
+:set:`toDescList` returns a list containing the elements of the set ``s`` in
+*descending* order.
+
+::
+
+    Set.toAscList (Set.fromList [0, 2, 4, 6])
+    > [0,2,4,6]
+
+    Set.toDescList (Set.fromList [0, 2, 4, 6]
+    > [6,4,2,0]
+
+
+Querying
+^^^^^^^^
+
+Check if a set is empty
+"""""""""""""""""""""""
+
+::
+
+    Set.null :: Set a -> Bool
+    Set.null s = ...
+
+:set:`null` returns ``True`` if the set ``s`` is empty, ``False`` otherwise.
+
+::
+
+    Set.null Set.empty
+    > True
+
+    Set.null (Set.fromList [0, 2, 4, 6])
+    > False
+
+
+The number of elements in a set
+"""""""""""""""""""""""""""""""
+
+::
+
+    Set.size :: Set a -> Int
+    Set.size s = ...
+
+:set:`size` returns the number of elements in the set ``s``.
+
+::
+
+    Set.size Set.empty
+    > 0
+
+    Set.size (Set.fromList [0, 2, 4, 6])
+    > 4
+
+Check if an element is in a set (member)
+""""""""""""""""""""""""""""""""""""""""
+
+::
+
+    Set.member :: Ord a => a -> Set a -> Bool
+    Set.member x s = ...
+
+:set:`member` returns ``True`` if the element ``x`` is in the set ``s``,
+``False`` otherwise.
+
+::
+
+    Set.member 0 Set.empty
+    > False
+
+    Set.member 0 (Set.fromList [0, 2, 4, 6])
+    > True
+
+Find the minimum/maximum element in a set
+"""""""""""""""""""""""""""""""""""""""""
+
+*Since version 0.5.9*
+
+::
+
+   lookupMin, lookupMax :: Set a -> Maybe a
+   lookupMin s = ...
+   lookupMax s = ...
+
+:set:`lookupMin` returns the minimum, or maximum respectively, element of the
+set ``s``, or ``Nothing`` if the set is empty.
+
+::
+
+    Set.lookupMin Set.empty
+    > Nothing
+
+    Set.lookupMin (Set.fromList [0, 2, 4, 6])
+    > Just 0
+
+    Set.lookupMax (Set.fromList [0, 2, 4, 6])
+    > Just 6
+
+.. WARNING::
+   Unless you're using an old version of ``containers`` **DO NOT** use
+   ``Set.findMin`` or ``Set.findMax``, they are partial and throw a runtime
+   error if the set is empty.
+
+Modification
+^^^^^^^^^^^^
+
+Adding a new element to a set
+"""""""""""""""""""""""""""""
+
+::
+
+    Set.insert :: Ord a => a -> Set a -> Set a
+    Set.insert x s = ...
+
+:set:`insert` places the element ``x`` into the set ``s``, replacing an existing
+equal element if it already exists.
+
+::
+
+    Set.insert 100 Set.empty
+    > fromList [100]
+
+    Set.insert 0 (Set.fromList [0, 2, 4, 6])
+    > fromList [0,2,4,6]
+
+Removing an element from a set
+""""""""""""""""""""""""""""""
+
+::
+
+    Set.delete :: Ord a => a -> Set a -> Set a
+    Set.delete x s = ...
+
+:set:`delete` the element ``x`` from the set ``s``, if it’s not a member it
+leaves the set unchanged.
+
+::
+
+    Set.delete 0 (Set.fromList [0, 2, 4, 6])
+    > fromList [2,4,6]
+
+Filtering elements from a set
+"""""""""""""""""""""""""""""
+
+::
+
+    Set.filter :: (a -> Bool) -> Set a -> Set a
+    Set.filter predicate s = ...
+
+:set:`filter` removes elements from the set ``s`` that **do not match** the
```

Comment:
Be positive!   `filter p s` produces a set consisting of the elements of `s` for which `p` returns `True`.

Reply:
Much clearer :)